### ISSUE TITLE
some issues about training

### ISSUE THREAD

**Issue by Seredris0622** [JinLi998/CoSMo3D#9]
Dear author, I have read your paper, and it is a well-written article. However, I have some questions. While training the code you provided, I noticed that the loss_weight might not be set correctly. Why is the avg_loss so high in the results I obtained? I would appreciate a detailed explanation

**Comment by Seredris0622** [comment]
When running the command `python eval_benchmark/eval_benchmark.py --benchmark 3dcompat200 --data_root dataset/3dcompat200 --checkpoint_path results/find3d_release/ckpt_final.pth --save_dir results/3dcompat200`, the error `No such file or directory: 'dataset/3dcompat200/air_conditioning_unit/label_map.json'` appears. Please provide a solution.

**Comment by JinLi998** [comment]
> Dear author, I have read your paper, and it is a well-written article. However, I have some questions. While training the code you provided, I noticed that the loss_weight might not be set correctly. Why is the avg_loss so high in the results I obtained? I would appreciate a detailed explanation

Thank you for your recognition of our work. To help us analyze the high average loss issue effectively, could you please provide more details? For instance, the current training epoch and the specific value of each individual loss term during training.

**Comment by JinLi998** [comment]
> When running the command `python eval_benchmark/eval_benchmark.py --benchmark 3dcompat200 --data_root dataset/3dcompat200 --checkpoint_path results/find3d_release/ckpt_final.pth --save_dir results/3dcompat200`, the error `No such file or directory: 'dataset/3dcompat200/air_conditioning_unit/label_map.json'` appears. Please provide a solution.

Thanks a lot for your feedback on our open-source code. The 3dcompat200 benchmark does not require the `label_map.json` file, which can be verified in the evaluation function `evaluate_3dcompat200` and the dataset loading class `Eval3dcom`.

It seems the program failed to call the correct `evaluate_3dcompat200` function. The script runs normally on our local environment. Could you please check your running environment and configurations? Feel free to share more details if you need further assistance.

**Comment by Seredris0622** [comment]
> > 尊敬的作者，我已阅读您的论文，文章写得很好。但我有一些疑问。在使用您提供的代码进行训练时，我注意到 loss_weight 的设置可能不正确。为什么我得到的结果中 avg_loss 的值这么高？希望能得到详细的解释。
> 
> 感谢您对我们工作的认可。为了帮助我们更有效地分析平均损失过高的问题，能否请您提供更多详细信息？例如，当前的训练轮数以及训练过程中每个损失项的具体数值。



> > Dear author, I have read your paper, and it is a well-written article. However, I have some questions. While training the code you provided, I noticed that the loss_weight might not be set correctly. Why is the avg_loss so high in the results I obtained? I would appreciate a detailed explanation
> 
> Thank you for your recognition of our work. To help us analyze the high average loss issue effectively, could you please provide more details? For instance, the current training epoch and the specific value of each individual loss term during training.

OK，could I provide you with my train.log?That may be workable.

**Comment by JinLi998** [comment]
> > > 尊敬的作者，我已阅读您的论文，文章写得很好。但我有一些疑问。在使用您提供的代码进行训练时，我注意到 loss_weight 的设置可能不正确。为什么我得到的结果中 avg_loss 的值这么高？希望能得到详细的解释。
> > 
> > 
> > 感谢您对我们工作的认可。为了帮助我们更有效地分析平均损失过高的问题，能否请您提供更多详细信息？例如，当前的训练轮数以及训练过程中每个损失项的具体数值。
> 
> > > Dear author, I have read your paper, and it is a well-written article. However, I have some questions. While training the code you provided, I noticed that the loss_weight might not be set correctly. Why is the avg_loss so high in the results I obtained? I would appreciate a detailed explanation
> > 
> > 
> > Thank you for your recognition of our work. To help us analyze the high average loss issue effectively, could you please provide more details? For instance, the current training epoch and the specific value of each individual loss term during training.
> 
> OK，could I provide you with my train.log?That may be workable.

Sure, you can send your train.log.

**Comment by Seredris0622** [comment]
> > > > 尊敬的作者，我已阅读您的论文，文章写得很好。但我有一些疑问。在使用您提供的代码进行训练时，我注意到 loss_weight 的设置可能不正确。为什么我得到的结果中 avg_loss 的值这么高？希望能得到详细的解释。
> > > 
> > > 
> > > 感谢您对我们工作的认可。为了帮助我们更有效地分析平均损失过高的问题，能否请您提供更多详细信息？例如，当前的训练轮数以及训练过程中每个损失项的具体数值。
> > 
> > 
> > > > Dear author, I have read your paper, and it is a well-written article. However, I have some questions. While training the code you provided, I noticed that the loss_weight might not be set correctly. Why is the avg_loss so high in the results I obtained? I would appreciate a detailed explanation
> > > 
> > > 
> > > Thank you for your recognition of our work. To help us analyze the high average loss issue effectively, could you please provide more details? For instance, the current training epoch and the specific value of each individual loss term during training.
> > 
> > 
> > OK，could I provide you with my train.log?That may be workable.
> 
> Sure, you can send your train.log.

[train.log](https://github.com/user-attachments/files/28423873/train.log)

**Comment by JinLi998** [comment]
> > > > > 尊敬的作者，我已阅读您的论文，文章写得很好。但我有一些疑问。在使用您提供的代码进行训练时，我注意到 loss_weight 的设置可能不正确。为什么我得到的结果中 avg_loss 的值这么高？希望能得到详细的解释。
> > > > 
> > > > 
> > > > 感谢您对我们工作的认可。为了帮助我们更有效地分析平均损失过高的问题，能否请您提供更多详细信息？例如，当前的训练轮数以及训练过程中每个损失项的具体数值。
> > > 
> > > 
> > > > > Dear author, I have read your paper, and it is a well-written article. However, I have some questions. While training the code you provided, I noticed that the loss_weight might not be set correctly. Why is the avg_loss so high in the results I obtained? I would appreciate a detailed explanation
> > > > 
> > > > 
> > > > Thank you for your recognition of our work. To help us analyze the high average loss issue effectively, could you please provide more details? For instance, the current training epoch and the specific value of each individual loss term during training.
> > > 
> > > 
> > > OK，could I provide you with my train.log?That may be workable.
> > 
> > 
> > Sure, you can send your train.log.
> 
> [train.log](https://github.com/user-attachments/files/28423873/train.log)




I've reviewed your log file, and the results are quite unusual.
The baseline loss is for contrastive learning; a value is excellent, indicating a strong alignment between semantic parts and text. The canon loss, which reflects the perception of canonicalization, also looks great.
However, the bbox loss shows the completely opposite result—the bounding boxes fail to enclose the parts properly.
Does it seem that the training was not performed on the dataset we provided? What are your settings for the learning rate and batch size? This doesn’t appear to be a simple parameter configuration issue.
I think checking the segmentation results on the training set would be an effective way to verify this.
If you’d like, we can communicate in real time via WeChat to improve efficiency. If you’re interested, please send your WeChat ID to my email: 202320851@mail.sdu.edu.cn.

**Comment by Seredris0622** [comment]
> > > > > > 尊敬的作者，我已阅读您的论文，文章写得很好。但我有一些疑问。在使用您提供的代码进行训练时，我注意到 loss_weight 的设置可能不正确。为什么我得到的结果中 avg_loss 的值这么高？希望能得到详细的解释。
> > > > > 
> > > > > 
> > > > > 感谢您对我们工作的认可。为了帮助我们更有效地分析平均损失过高的问题，能否请您提供更多详细信息？例如，当前的训练轮数以及训练过程中每个损失项的具体数值。
> > > > 
> > > > 
> > > > > > Dear author, I have read your paper, and it is a well-written article. However, I have some questions. While training the code you provided, I noticed that the loss_weight might not be set correctly. Why is the avg_loss so high in the results I obtained? I would appreciate a detailed explanation
> > > > > 
> > > > > 
> > > > > Thank you for your recognition of our work. To help us analyze the high average loss issue effectively, could you please provide more details? For instance, the current training epoch and the specific value of each individual loss term during training.
> > > > 
> > > > 
> > > > OK，could I provide you with my train.log?That may be workable.
> > > 
> > > 
> > > Sure, you can send your train.log.
> > 
> > [train.log](https://github.com/user-attachments/files/28423873/train.log)
> 
> 
> 
> 
> I've reviewed your log file, and the results are quite unusual.
> The baseline loss is for contrastive learning; a value is excellent, indicating a strong alignment between semantic parts and text. The canon loss, which reflects the perception of canonicalization, also looks great.
> However, the bbox loss shows the completely opposite result—the bounding boxes fail to enclose the parts properly.
> Does it seem that the training was not performed on the dataset we provided? What are your settings for the learning rate and batch size? This doesn’t appear to be a simple parameter configuration issue.
> I think checking the segmentation results on the training set would be an effective way to verify this.
> If you’d like, we can communicate in real time via WeChat to improve efficiency. If you’re interested, please send your WeChat ID to my email: 202320851@mail.sdu.edu.cn.

Sure,I will do that.Thank you for solving my problem patiently,author.

**Comment by Seredris0622** [comment]
> > > > > > 尊敬的作者，我已阅读您的论文，文章写得很好。但我有一些疑问。在使用您提供的代码进行训练时，我注意到 loss_weight 的设置可能不正确。为什么我得到的结果中 avg_loss 的值这么高？希望能得到详细的解释。
> > > > > 
> > > > > 
> > > > > 感谢您对我们工作的认可。为了帮助我们更有效地分析平均损失过高的问题，能否请您提供更多详细信息？例如，当前的训练轮数以及训练过程中每个损失项的具体数值。
> > > > 
> > > > 
> > > > > > Dear author, I have read your paper, and it is a well-written article. However, I have some questions. While training the code you provided, I noticed that the loss_weight might not be set correctly. Why is the avg_loss so high in the results I obtained? I would appreciate a detailed explanation
> > > > > 
> > > > > 
> > > > > Thank you for your recognition of our work. To help us analyze the high average loss issue effectively, could you please provide more details? For instance, the current training epoch and the specific value of each individual loss term during training.
> > > > 
> > > > 
> > > > OK，could I provide you with my train.log?That may be workable.
> > > 
> > > 
> > > Sure, you can send your train.log.
> > 
> > [train.log](https://github.com/user-attachments/files/28423873/train.log)
> 
> 
> 
> 
> I've reviewed your log file, and the results are quite unusual.
> The baseline loss is for contrastive learning; a value is excellent, indicating a strong alignment between semantic parts and text. The canon loss, which reflects the perception of canonicalization, also looks great.
> However, the bbox loss shows the completely opposite result—the bounding boxes fail to enclose the parts properly.
> Does it seem that the training was not performed on the dataset we provided? What are your settings for the learning rate and batch size? This doesn’t appear to be a simple parameter configuration issue.
> I think checking the segmentation results on the training set would be an effective way to verify this.
> If you’d like, we can communicate in real time via WeChat to improve efficiency. If you’re interested, please send your WeChat ID to my email: 202320851@mail.sdu.edu.cn.

Dear author, your email address seems to be wrong. I

**Comment by JinLi998** [comment]
> > > > > > > 尊敬的作者，我已阅读您的论文，文章写得很好。但我有一些疑问。在使用您提供的代码进行训练时，我注意到 loss_weight 的设置可能不正确。为什么我得到的结果中 avg_loss 的值这么高？希望能得到详细的解释。
> > > > > > 
> > > > > > 
> > > > > > 感谢您对我们工作的认可。为了帮助我们更有效地分析平均损失过高的问题，能否请您提供更多详细信息？例如，当前的训练轮数以及训练过程中每个损失项的具体数值。
> > > > > 
> > > > > 
> > > > > > > Dear author, I have read your paper, and it is a well-written article. However, I have some questions. While training the code you provided, I noticed that the loss_weight might not be set correctly. Why is the avg_loss so high in the results I obtained? I would appreciate a detailed explanation
> > > > > > 
> > > > > > 
> > > > > > Thank you for your recognition of our work. To help us analyze the high average loss issue effectively, could you please provide more details? For instance, the current training epoch and the specific value of each individual loss term during training.
> > > > > 
> > > > > 
> > > > > OK，could I provide you with my train.log?That may be workable.
> > > > 
> > > > 
> > > > Sure, you can send your train.log.
> > > 
> > > 
> > > [train.log](https://github.com/user-attachments/files/28423873/train.log)
> > 
> > 
> > I've reviewed your log file, and the results are quite unusual.
> > The baseline loss is for contrastive learning; a value is excellent, indicating a strong alignment between semantic parts and text. The canon loss, which reflects the perception of canonicalization, also looks great.
> > However, the bbox loss shows the completely opposite result—the bounding boxes fail to enclose the parts properly.
> > Does it seem that the training was not performed on the dataset we provided? What are your settings for the learning rate and batch size? This doesn’t appear to be a simple parameter configuration issue.
> > I think checking the segmentation results on the training set would be an effective way to verify this.
> > If you’d like, we can communicate in real time via WeChat to improve efficiency. If you’re interested, please send your WeChat ID to my email: [202320851@mail.sdu.edu.cn](mailto:202320851@mail.sdu.edu.cn).
> 
> I can't send you an email.

My mistake for writing the wrong email, the right address is: jinli@mail.sdu.edu.cn
